### PR TITLE
KOGITO-3711: BDD tests: Unable to find latest tagged image

### PR DESCRIPTION
**Pull request by Sgitario**
JIRA: https://issues.redhat.com/browse/KOGITO-3711
Description: Build <runtime> binary build with native <native> using KogitoBuild" is invoking binary build using OC client. Sometimes this build crash due to "unable to find latest tagged image".

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by Sgitario** [comment]
This issue is happening at the step "Step Start build with name "YY" from local example service path XX" which internally is using:

```
oc start-build YYY ...
```

As stated in the jira ticket, the above OC command is failing when the image stream is not ready yet. Then, I could have resolved this issue by checking whether the image stream is created beforehand. However, I discarded this solution as doing the retry was significantly easier to implement. (As it avoids to check how the build was created or get the required image stream to check...).
Let me know if this solution is ok for you. I did verify that this is working fine in the worst scenario (the image stream can't find the docker image), and the best scenario (everything works at first).

**Review by sutaakar — APPROVED** [review]
Looks good to me.
Thanks for a quick fix.

File: test/steps/openshift.go [hunk 1]
```diff
@@ -42,14 +46,20 @@ func registerOpenShiftSteps(ctx *godog.ScenarioContext, data *Data) {
 // Build steps
 func (data *Data) startBuildFromExampleServicePath(buildName, localExamplePath string) error {
 	examplesRepositoryPath := data.KogitoExamplesLocation
-	_, err := framework.CreateCommand("oc", "start-build", buildName, "--from-dir="+examplesRepositoryPath+"/"+localExamplePath, "-n", data.Namespace).WithLoggerContext(data.Namespace).Execute()
-	return err
+	return framework.WaitForOnOpenshift(data.Namespace, fmt.Sprintf("Build '%s' to start", buildName), defaultTimeoutToStartBuildInMin,
+		func() (bool, error) {
+			_, err := framework.CreateCommand("oc", "start-build", buildName, "--from-dir="+examplesRepositoryPath+"/"+localExamplePath, "-n", data.Namespace).WithLoggerContext(data.Namespace).Execute()
```
Comment (ricardozanini) [inline test/steps/openshift.go:51]: Hey @Sgitario!

Quick question: while do we need to manually start the build like this?

Comment (Sgitario) [reply test/steps/openshift.go:51]: It's needed until https://issues.redhat.com/browse/KOGITO-2161 is done. In this scenario, we're deploying from binaries (more in [this comment](https://issues.redhat.com/browse/KOGITO-2161?focusedCommentId=14159830&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-14159830)).

**Review by radtriste — APPROVED** [review]
lgtm

File: test/steps/openshift.go (hunk 1 above)
Comment (radtriste) [reply test/steps/openshift.go:51]: @Sgitario Can you just add the issue in comment ?

Comment (Sgitario) [reply test/steps/openshift.go:51]: The issue was already added by Karel in the scenario as a comment.

**Pull request merged**